src/ds3.c

```diff
@@ -1287,19 +1293,29 @@ static ds3_str* xml_get_string(xmlDocPtr doc, xmlNodePtr child_node) {
     xmlChar* text;
     ds3_str* result;
     text = xmlNodeListGetString(doc, child_node->xmlChildrenNode, 1);
+    if (NULL == text){
+        // Element is found, but is empty: <name />
+        return NULL;
+    }
     result = ds3_str_init((const char*) text);
     xmlFree(text);
     return result;
 }
 
+/*
```

Comment:
This commented code should be removed unless there is a reason why we should keep it.